wavefront/server/plugins/datasource/datasource/redshift/__init__.py

```
@@ -63,9 +72,67 @@ def execute_dynamic_query(
-        # TODO: Implement RLS filter support for Redshift
-        # For now, just execute the query without RLS filter
-        pass
+        results = {}
+        tasks = []
+
+        for query_obj in query:
+            query_to_execute = query_obj.get('query', '')
+            query_params = query_obj.get('parameters', {})
+            query_id = query_obj.get('id')
+            if not query_id:
+                raise ValueError('Query ID is required')
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)